wasm2c/wasm-rt-mem-impl.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2018 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "wasm-rt-impl.h"
+
+#include <assert.h>
+#include <stdio.h>
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/mman.h>
+#endif
+
+#define PAGE_SIZE 65536
+
+#ifdef WASM_RT_GROW_FAILED_HANDLER
+extern void WASM_RT_GROW_FAILED_HANDLER();
+#endif
+
+#define C11_MEMORY_LOCK_VAR_INIT(name)                \
+  if (mtx_init(&(name), mtx_plain) != thrd_success) { \
+    fprintf(stderr, "Lock init failed\n");            \
+    abort();                                          \
+  }
+#define C11_MEMORY_LOCK_AQUIRE(name)          \
+  if (mtx_lock(&(name)) != thrd_success) {    \
+    fprintf(stderr, "Lock acquire failed\n"); \
+    abort();                                  \
+  }
+#define C11_MEMORY_LOCK_RELEASE(name)         \
+  if (mtx_unlock(&(name)) != thrd_success) {  \
+    fprintf(stderr, "Lock release failed\n"); \
+    abort();                                  \
+  }
+
+#define PTHREAD_MEMORY_LOCK_VAR_INIT(name)      \
+  if (pthread_mutex_init(&(name), NULL) != 0) { \
+    fprintf(stderr, "Lock init failed\n");      \
+    abort();                                    \
+  }
+#define PTHREAD_MEMORY_LOCK_AQUIRE(name)      \
+  if (pthread_mutex_lock(&(name)) != 0) {     \
+    fprintf(stderr, "Lock acquire failed\n"); \
+    abort();                                  \
+  }
+#define PTHREAD_MEMORY_LOCK_RELEASE(name)     \
+  if (pthread_mutex_unlock(&(name)) != 0) {   \
+    fprintf(stderr, "Lock release failed\n"); \
```

Comment:
Should we have another macro here for LOG_FAILURE, that doesn't nothing in NDEBUG builder?

Reply:
I think for cases that cause a abort/crash, I am generally in favor of having the message always print. So my preference would be to keep this as is unless you feel strongly.

Reply:
I don't feel strongly.  If someone really cares about code size they could remove them..